sw/device/lib/testing/dif_uart_test.cc

```diff
@@ -0,0 +1,819 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+
+extern "C" {
+#include "sw/device/lib/dif/dif_uart.h"
+#include "uart_regs.h"  // Generated.
+}  // extern "C"
+
+namespace {
+  class DifUartTest : public testing::Test, public mock_mmio::MmioTest {
+ protected:
+    dif_uart_t dif_uart_;
+
+    // NCO is calculated as NCO = 2^20 * fbaud / fpclk, so the following values
+    // should result in NCO of 1.
+    dif_uart_config_t dif_uart_config_ = {
+      1,                  // baudrate
+      1048576,            // clk_freq_hz
+      kDifUartDisable,    // parity_enable
+      kDifUartParityEven, // parity
+    };
+};
+
+class DifUartTestUartInitialised : public DifUartTest {
```

Comment:
Avoid creating fixtures just for a single behavior. Saving a single line in every test is not worth creating an entire fixture.

One option is to add a member function to DifUartTest along the lines of `InitDevice()` that does what this class's constructor does.